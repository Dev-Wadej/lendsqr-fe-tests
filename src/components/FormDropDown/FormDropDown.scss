    .form-dropdown {
        position: absolute;
        top: 3.8rem;
        width: 17rem;
        background: #FFFFFF;
        border: 1px solid rgba(84, 95, 125, 0.14);
        box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
        border-radius: .4rem;
        color: rgba(84, 95, 125, 1);
        font-family: 'Work Sans', sans-serif;
        font-size: .87rem;
        font-weight: 500;
        padding-top: 1.5rem;
        padding-left: 1rem;
        padding-bottom: 1.7rem;
        z-index: 2000;
        div {
            margin-bottom: 1rem;
            label,
            input,
            select {
                display: block;
            }
            input,
            select {
                border: 1px solid #d6e1f1;
                border-radius: .4rem;
                width: 14rem;
                margin-top: .4rem;
                padding: .5rem;
                cursor: pointer;
                &::placeholder {
                    color: #545F7D;
                    opacity: 0.7;
                    font-size: .77rem;
                    padding-left: .6rem;
                }
            }
        }
    }
    
    select:focus {
        outline: none;
    }
    
     ::-webkit-datetime-edit-fields-wrapper {
        color: #545F7D;
        opacity: .7;
    }
    
     ::-webkit-datetime-edit-text {
        color: #545F7D;
        opacity: .7;
    }
    
     ::-webkit-datetime-edit-month-field {
        color: #545F7D;
        opacity: .7;
    }
    
     ::-webkit-datetime-edit-day-field {
        color: #545F7D;
        opacity: .7;
    }
    
     ::-webkit-datetime-edit-year-field {
        color: #545F7D;
        opacity: .7;
    }
    
    .form-btns {
        display: flex;
        gap: 1rem;
        button {
            display: inline-block;
            font-family: 'Work Sans', sans-serif;
        }
        button[type='submit'] {
            background: #39CDCC;
            border: 1px solid #39CDCC;
            border-radius: .4rem;
            padding: .5rem 1.8rem;
            color: #fff;
        }
        button[type='reset'] {
            border: 1px solid #545F7D;
            border-radius: 8px;
            color: rgba(84, 95, 125, 1);
            padding: .5rem 1.8rem;
        }
    }