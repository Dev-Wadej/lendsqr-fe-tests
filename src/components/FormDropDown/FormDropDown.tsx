import React from 'react'
import './FormDropDown.scss';

const FormDropDown = () => {

    const handleFilter=(e:React.FormEvent<HTMLButtonElement>)=>{
        e.preventDefault()
    }
  return (
    <form className='form-dropdown'>
        
        <div>
            <label htmlFor="organization" >Organization</label>
            <select name="organization" placeholder='Select' id="organization">
            <option  value="select">Select</option>
        </select>
           
        </div>
        <div>
            <label htmlFor="username" >Username</label>
            <input type="text" placeholder='User' id='username' />
        </div>
        <div>
            <label htmlFor="email" >Email</label>
            <input type="email" placeholder='Email' id='email' />
        </div>
        <div>
            <label htmlFor="date" >Date</label>
            <input type="date" placeholder='Date' id='date'/>
        </div>
        <div>
            <label htmlFor="number" >Phone Number</label>
            <input type="number" placeholder='Phone Number' id='number'/>
        </div>
        <div>
            <label htmlFor="organization" >Organization</label>
            <select name="organization" placeholder='Select' id="organization">
            <option value="select">Select</option>
        </select>
           
        </div>
        <br />
        <div className='form-btns'>

        <button type="reset">Reset</button>
        <br />
        <button onClick={handleFilter} type="submit">Filter</button>
        </div>
      
    </form>
  )
}

export default FormDropDown
