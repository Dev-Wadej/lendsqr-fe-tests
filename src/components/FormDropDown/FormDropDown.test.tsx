import FormDropDown from './FormDropDown'
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';



afterEach(cleanup)
test('Form component snapshot HTML', () => {
  const output = renderer.create(<FormDropDown/>).toJSON();
  expect(output).toMatchSnapshot();
});