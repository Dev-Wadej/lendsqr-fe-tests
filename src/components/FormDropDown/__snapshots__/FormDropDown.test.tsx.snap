// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot HTML 1`] = `
<form
  className="form-dropdown"
>
  <div>
    <label
      htmlFor="organization"
    >
      Organization
    </label>
    <select
      id="organization"
      name="organization"
      placeholder="Select"
    >
      <option
        value="select"
      >
        Select
      </option>
    </select>
  </div>
  <div>
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      id="username"
      placeholder="User"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      placeholder="Email"
      type="email"
    />
  </div>
  <div>
    <label
      htmlFor="date"
    >
      Date
    </label>
    <input
      id="date"
      placeholder="Date"
      type="date"
    />
  </div>
  <div>
    <label
      htmlFor="number"
    >
      Phone Number
    </label>
    <input
      id="number"
      placeholder="Phone Number"
      type="number"
    />
  </div>
  <div>
    <label
      htmlFor="organization"
    >
      Organization
    </label>
    <select
      id="organization"
      name="organization"
      placeholder="Select"
    >
      <option
        value="select"
      >
        Select
      </option>
    </select>
  </div>
  <br />
  <div
    className="form-btns"
  >
    <button
      type="reset"
    >
      Reset
    </button>
    <br />
    <button
      onClick={[Function]}
      type="submit"
    >
      Filter
    </button>
  </div>
</form>
`;
